signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/EthPublicKeyUtils.java

```diff
@@ -12,63 +12,254 @@
-import java.security.spec.InvalidParameterSpecException;
+import java.util.Arrays;
 
 import org.apache.tuweni.bytes.Bytes;
-import org.apache.tuweni.bytes.Bytes32;
-import org.web3j.utils.Numeric;
+import org.bouncycastle.asn1.x9.X9ECParameters;
+import org.bouncycastle.crypto.ec.CustomNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPublicKey;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jce.spec.ECParameterSpec;
+import org.bouncycastle.math.ec.ECCurve;
+import org.bouncycastle.math.ec.ECPoint;
+import org.web3j.crypto.ECKeyPair;
 
+/**
+ * Utility class for working with secp256k1 public keys. This class provides methods for converting
+ * between Java and Web3J library based SECP keys.
+ */
 public class EthPublicKeyUtils {
-  private static final int PUBLIC_KEY_SIZE = 64;
+  private static final BouncyCastleProvider BC_PROVIDER = new BouncyCastleProvider();
+  private static final ECDomainParameters SECP256K1_DOMAIN;
+  private static final ECParameterSpec BC_SECP256K1_SPEC;
+  private static final java.security.spec.ECParameterSpec JAVA_SECP256K1_SPEC;
+  private static final String SECP256K1_CURVE = "secp256k1";
+  private static final String EC_ALGORITHM = "EC";
+
+  static {
+    final X9ECParameters params = CustomNamedCurves.getByName(SECP256K1_CURVE);
+    SECP256K1_DOMAIN =
+        new ECDomainParameters(params.getCurve(), params.getG(), params.getN(), params.getH());
+    BC_SECP256K1_SPEC =
+        new ECParameterSpec(params.getCurve(), params.getG(), params.getN(), params.getH());
+    final ECCurve bcCurve = BC_SECP256K1_SPEC.getCurve();
+    JAVA_SECP256K1_SPEC =
+        new java.security.spec.ECParameterSpec(
+            new EllipticCurve(
+                new java.security.spec.ECFieldFp(bcCurve.getField().getCharacteristic()),
+                bcCurve.getA().toBigInteger(),
+                bcCurve.getB().toBigInteger()),
+            new java.security.spec.ECPoint(
+                BC_SECP256K1_SPEC.getG().getAffineXCoord().toBigInteger(),
+                BC_SECP256K1_SPEC.getG().getAffineYCoord().toBigInteger()),
+            BC_SECP256K1_SPEC.getN(),
+            BC_SECP256K1_SPEC.getH().intValue());
+  }
+
+  /**
+   * Create a new secp256k1 key pair.
+   *
+   * @param random The random number generator to use
+   * @return The generated java security key pair
+   * @throws GeneralSecurityException If there is an issue generating the key pair
+   */
+  public static KeyPair createSecp256k1KeyPair(final SecureRandom random)
+      throws GeneralSecurityException {
+    final KeyPairGenerator keyPairGenerator =
+        KeyPairGenerator.getInstance(EC_ALGORITHM, BC_PROVIDER);
+    final ECGenParameterSpec ecGenParameterSpec = new ECGenParameterSpec(SECP256K1_CURVE);
+    if (random != null) {
+      keyPairGenerator.initialize(ecGenParameterSpec, random);
```

Comment:
why wouldn't we always create the key pair using a securerandom?

Reply:
I think my original intent was that we might send in null SecureRandom. I am going to remove this check as we will always send-in an initialize SecureRandom class.

Reply:
Updated to use a static final instance of SecureRandom in this class.